python/nammu/controller/AtfAreaController.py

```diff
@@ -122,6 +122,35 @@ def set_validation_errors(self, validation_errors):
         '''
         self.validation_errors = validation_errors
 
+    def undoOrRedo(self, forward):
+        """
+        Helper function for undo() and redo().  The boolean argument `forward`
+        specifies if the edit is to be undone (`False`) or redone (`True`).
+        """
+        currentEdit = None
+        try:
+            # Before actually redoing/undoing, get the edit about to be
+            # redone/undone, in order to then move focus to its pane.
+            if forward:
+                currentEdit = self.undo_manager.editToBeRedone()
+                self.undo_manager.redo()
+            else:
+                currentEdit = self.undo_manager.editToBeUndone()
+                self.undo_manager.undo()
+        except (CannotUndoException, CannotRedoException):
+            # These exceptions indicate we've reached the end of the edits
+            # vector.  Nothing to do
+            pass
+        else:
+            if currentEdit:
```

Comment:
I don't think that this `if` is necessary. It seems to me that `currentEdit` is only falsy (`None`) when there are no edits to undo/redo, but this will have caused an exception further up, so we won't be in this block. I've tried it a bit and nothing crashes, for whatever that's worth...
I don't mind keeping the check here for the sake of safety (in case there is something else we haven't foreseen), but if so it would be good to put a comment that we don't expect to be here.